pytorch_tabnet/tab_network.py

```diff
@@ -23,37 +23,38 @@ class GBN(torch.nn.Module):
         Ghost Batch Normalization
         https://arxiv.org/abs/1705.08741
     """
-    def __init__(self, input_dim, virtual_batch_size=128, momentum=0.01, device='cpu'):
+
+    def __init__(self, input_dim, virtual_batch_size=128, momentum=0.01):
         super(GBN, self).__init__()
 
         self.input_dim = input_dim
         self.virtual_batch_size = virtual_batch_size
         self.bn = BatchNorm1d(self.input_dim, momentum=momentum)
-        self.device = device
 
     def forward(self, x):
         chunks = x.chunk(x.shape[0] // self.virtual_batch_size +
                          ((x.shape[0] % self.virtual_batch_size) > 0))
-        res = torch.Tensor([]).to(self.device)
+        res = torch.Tensor([]).to(x.device)
         for x_ in chunks:
             y = self.bn(x_)
             res = torch.cat([res, y], dim=0)
 
         return res
 
 
-class TabNet(torch.nn.Module):
-    def __init__(self, input_dim, output_dim, n_d=8, n_a=8,
-                 n_steps=3, gamma=1.3, cat_idxs=[], cat_dims=[], cat_emb_dim=1,
+class DryTabNet(torch.nn.Module):
```

Comment:
could you rename `DryTabNet` into `TabNetNoEmbeddings`? I think it's a bit more expressive that way.

Reply:
thx!